### PR TITLE
WIP: Include caviar V1

### PR DESCRIPTION
**Pull request by outdoteth**

### CODE REVIEW

File: packages/sdk/src/router/v6/addresses.ts [hunk 1]
```diff
@@ -121,6 +121,11 @@ export const SudoswapV2Module: ChainIdToAddress = {
   [Network.EthereumGoerli]: "0xe2840826c43c25e88a5ef43ff790d7105889dd6e",
 };
 
+export const CaviarV1Module: ChainIdToAddress = {
+  [Network.Ethereum]: "todo insert here",
+  [Network.EthereumGoerli]: "todo insert here",
```
Comment (outdoteth) [inline packages/sdk/src/router/v6/addresses.ts:126]: when should we deploy these? after the PR is merged or beforehand?

File: packages/sdk/src/router/v6/router.ts [hunk 2]
```diff
@@ -3477,6 +3554,36 @@ export class Router {
...
+          executionsWithDetails.push({
+            detail,
+            execution: {
+              module: module.address,
+              data: module.interface.encodeFunctionData("sell", [
+                order.params.pool,
+                detail.tokenId,
+                bn(order.params.extra.prices[0]),
+                // todo: should this stolen nft oracle message be fetched here or on the client side
+                { id: constants.HashZero, payload: [], timestamp: 0, signature: [] },
```
Comment (outdoteth) [inline packages/sdk/src/router/v6/router.ts:3570]: in caviar we verify that all NFTs that are sold into the pools are not marked as stolen by reservoir's oracle.
should the stolen NFT attestation from the oracle be fetched on the server here, or should it be passed from the client?

File: packages/indexer/src/common/logger.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ const log = (level: "error" | "info" | "warn" | "debug") => {
 
   const logger = createLogger({
     exitOnError: false,
-    level: "debug",
+    level: "info",
```
Comment (georgeroman) [inline packages/indexer/src/common/logger.ts:28]: Unrelated change, let's leave this as it was

File: packages/indexer/src/sync/events/handlers/utils/index.ts [hunk 4]
```diff
@@ -94,6 +94,9 @@ export const initOnChainData = (): OnChainData => ({
 export const processOnChainData = async (data: OnChainData, backfill?: boolean) => {
   // Post-process fill events
 
+  // TODO: remove this, only here for testing
```
Comment (georgeroman) [inline packages/indexer/src/sync/events/handlers/utils/index.ts:97]: Leftover?

File: packages/sdk/src/caviar-v1/addresses.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { ChainIdToAddress, Network } from "../utils";
+
+export const CaviarFactoryContract: ChainIdToAddress = {
+  [Network.Ethereum]: "0xa964D6E8d90E5Cd12592a8EF2b1735daE9Ba0840",
```
Comment (georgeroman) [inline packages/sdk/src/caviar-v1/addresses.ts:4]: Addresses should be lowercase

File: packages/sdk/src/caviar-v1/order.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+const normalize = (order: Types.OrderParams): Types.OrderParams => {
+  // Perform some normalization operations on the order:
+  // - convert bignumbers to strings where needed
+  // - convert strings to numbers where needed
+  // - lowercase all strings
+
+  return {
+    pool: lc(order.pool),
+    tokenId: order.tokenId,
```
Comment (georgeroman) [inline packages/sdk/src/caviar-v1/order.ts:27]: For safety, let's stringify this: `order.tokenId ? s(order.tokenId) : order.tokenId`

File: packages/sdk/src/router/v6/addresses.ts (hunk 1 above)
Comment (georgeroman) [reply packages/sdk/src/router/v6/addresses.ts:126]: Before, I'll take care of deploying before merging

File: packages/sdk/src/router/v6/router.ts (hunk 2 above)
Comment (georgeroman) [reply packages/sdk/src/router/v6/router.ts:3570]: I think it should be fetched from the server. See how it's done below, in the `collectionxyz` case (we use the `extraArgs` field).

File: packages/sdk/src/router/v6/router.ts [hunk 7]
```diff
@@ -56,6 +56,8 @@ import X2Y2ModuleAbi from "./abis/X2Y2Module.json";
 import ZeroExV4ModuleAbi from "./abis/ZeroExV4Module.json";
 import ZoraModuleAbi from "./abis/ZoraModule.json";
 import SudoswapV2ModuleAbi from "./abis/SudoswapV2Module.json";
+import CaviarV1ModuleAbi from "./abis/CaviarV1Module.json";
+import { constants } from "ethers";
```
Comment (georgeroman) [inline packages/sdk/src/router/v6/router.ts:60]: We already import `HashZero`

**Review by georgeroman — CHANGES_REQUESTED** [review]

File: packages/sdk/src/caviar-v1/exchange.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+export class Exchange {
+  public async fetchStolenProof(tokenId: string, tokenAddress: string) {
+    const reservoirUrl = `https://api.reservoir.tools/oracle/tokens/status/v2?tokens=${tokenAddress}:${tokenId}}`;
+
+    const { messages } = await fetch(reservoirUrl, {
+      headers: { "x-api-key": process.env.ORACLE_API_KEY as string },
```
Comment (outdoteth) [inline packages/sdk/src/caviar-v1/exchange.ts:6]: is this the correct way to pass in the env var?

File: packages/sdk/src/router/v6/router.ts (hunk 2 above)
Comment (outdoteth) [reply packages/sdk/src/router/v6/router.ts]: added an Exchange class that has a method to fetch the stolen proof.

File: packages/sdk/src/caviar-v1/exchange.ts (hunk 8 above)
Comment (georgeroman) [reply packages/sdk/src/caviar-v1/exchange.ts:6]: We should probably use `inject` instead: https://github.com/reservoirprotocol/indexer/blob/ae578a3613044a6e14e537b28b8d3ceb15386a04/packages/indexer/src/api/endpoints/management/post-simulate-order/v1.ts#L169-L182

Also, we should pass the signature to the SDK via the `extraArgs` field, as below: https://github.com/reservoirprotocol/indexer/blob/ae578a3613044a6e14e537b28b8d3ceb15386a04/packages/indexer/src/orderbook/orders/index.ts#L482-L499

File: packages/indexer/src/orderbook/orders/index.ts [hunk 9]
```diff
@@ -753,6 +766,29 @@ export const generateBidDetailsV6 = async (
...
+    case "caviar-v1": {
+      const sdkOrder = new Sdk.CaviarV1.Order(config.chainId, order.rawData);
+
+      const response = await inject({
+        method: "GET",
+        url: `/oracle/tokens/status/v2?tokens=${sdkOrder.params.tokenAddress}:${sdkOrder.params.tokenId}}`,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
```
Comment (outdoteth) [inline packages/indexer/src/orderbook/orders/index.ts:778]: @georgeroman like this?

File: packages/sdk/src/caviar-v1/exchange.ts (hunk 8 above)
Comment (outdoteth) [reply packages/sdk/src/caviar-v1/exchange.ts]: cool, moved it around. got rid of the exchange and switched to inject and extraArgs

**Pull request merged**